MvvmCross/Platforms/Uap/Presenters/MvxWindowsViewPresenter.cs

```diff
@@ -240,5 +251,36 @@ protected virtual Task<bool> ShowPage(Type viewType, MvxBasePresentationAttribut
                 return Task.FromResult(false);
             }
         }
+
+        protected virtual async Task<bool> ShowModal(Type viewType, MvxModalViewPresentationAttribute attribute, MvxViewModelInstanceRequest request)
+        {
+            try
+            {
+                if (Activator.CreateInstance(viewType) is MvxWindowsContentDialog modalView)
+                {
+                    modalView.ViewModel = request.ViewModelInstance;
+
+                    await modalView.ShowAsync(attribute.Placement);
+
+                    return true;
+                }
+
+                return false;
+            }
+            catch (Exception exception)
+            {
+                MvxLog.Instance.Trace("Error seen during navigation request to {0} - error {1}", request.ViewModelType.Name,
+                    exception.ToLongString());
+                return false;
+            }
+        }
+
+        protected virtual Task<bool> CloseModal(IMvxViewModel viewModel, MvxBasePresentationAttribute attribute)
+        {
+            var popups = VisualTreeHelper.GetOpenPopups(Window.Current).FirstOrDefault(p => p.Child is ContentDialog);
```

Comment:
Check for `MvxWindowsContentDialog` to get a closer match. If possible also check the viewmodel type or something to make sure we don't close the wrong one when multiple modals are open.

Reply:
You cannot open two ContentDialog at a time.